connection.go

```diff
@@ -100,19 +96,11 @@ var (
 	// on a closed connection
 	ErrConnectionClosed = errors.New("connection is closed")
 
-	// ErrConnectionNotReady is returned when a caller attempts to send a
-	// request through a connection which has not yet been initialized
-	ErrConnectionNotReady = errors.New("connection is not yet ready")
```

Comment:
Removing this is technically a breaking change. Can we keep it and just document that it's now unused?